probe-rs/src/core/mod.rs

```diff
@@ -411,6 +417,20 @@ impl SpecificCoreState {
                 crate::architecture::arm::armv7m::Armv7m::new(memory, s, debug_sequence)?,
                 state,
             ),
+            SpecificCoreState::Armv8a(s) => Core::new(
+                crate::architecture::arm::armv8a::Armv8a::new(
+                    memory,
+                    s,
+                    base_address.ok_or_else(|| {
```

Comment:
I think we should make this a panic and not a dynamic error. And instead we need to introduce a validation for the target yaml where we load it to make sure the yaml contains the base address and the cti address.
We do this for flash algorithms too. This way we can make sure that we do not drag errors to places where they do not make much sense and just clutter the API.

Reply:
Sure thing - moved the validation to ChipFamily's validate method and put expect calls in the core code.